src/containerapp/azext_containerapp/custom.py

```diff
@@ -523,12 +532,19 @@ def update_containerapp_logic(cmd,
         containerapp_def = ContainerAppClient.show(cmd=cmd, resource_group_name=resource_group_name, name=name)
     except:
         pass
-
     if not containerapp_def:
         raise ResourceNotFoundError("The containerapp '{}' does not exist".format(name))
 
     new_containerapp = {}
     new_containerapp["properties"] = {}
+    if source and not registry_server:
+        registry_server = containerapp_def["properties"]["configuration"]["registries"][0]["server"]
```

Comment:
same as above message, do we know that this will not throw errors from being unable to access any of these properties?

Reply:
It is a requirement to provide registry-server, username and password with source/repo with create. So this `containerapp_def["properties"]["configuration"]["registries"][0]["server"]` will always be a non-null value.